### PR TITLE
Kaniko proxy

### PR DESCRIPTION
**Pull request by prary**
Implementation of PR #2064 Setting proxy for Kaniko Pod

### CODE REVIEW

**Comment by prary** [comment]
@balopat fixed travis error.

File: pkg/skaffold/build/cluster/sources/sources.go [hunk 1]
```diff
@@ -135,6 +138,23 @@ func podTemplate(clusterDetails *latest.ClusterDetails, artifact *latest.KanikoA
 	return pod
 }
 
+func setProxy(clusterDetails *latest.ClusterDetails, env []v1.EnvVar) []v1.EnvVar {
```
Comment (tejal29) [inline pkg/skaffold/build/cluster/sources/sources.go]: Can we add tests for setProxy with following test cases
1. clusterDetails has only HTTPProxy set
2. clusterDetails has only HTTPSProxy set
3. clusterDetails has  HTTPProxy and HTTPSProxy set
4. clusterDetails has nothing set returns nothing

Comment (prary) [reply pkg/skaffold/build/cluster/sources/sources.go]: Sure will do that.

**Comment by tejal29** [comment]
please discard this if #2352 works for you

File: pkg/skaffold/build/cluster/sources/sources.go (hunk 1 above)
Comment (prary) [reply pkg/skaffold/build/cluster/sources/sources.go]: added

**Comment by priyawadhwa** [comment]
Hey @prary , please run `make generate-schemas` and commit the results for Travis to pass!

**Pull request merged**